opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -153,6 +157,20 @@ public Thread newThread(Runnable runnable) {
             });
     }
 
+    public ExecutorService getRevisionExecutor() {
+        return lzRevisionExecutor.get();
+    }
+
+    private ExecutorService newRevisionExecutor() {
+        return Executors.newFixedThreadPool(this.getMaxRevisionThreadCount(),
+                new NamedThreadFactory("get-revision"));
+    }
+
+    public void shutdownRevisionExecutor() throws InterruptedException {
+        getRevisionExecutor().shutdownNow();
+        getRevisionExecutor().awaitTermination(0, TimeUnit.SECONDS);
```

Comment:
Is there an effective difference between awaiting `0` or just not awaiting?

Reply:
for some reason I thought this will wait indefinitely. Will use some sane value.